server/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -220,7 +220,8 @@ public void quickExec(String statement, ResultReceiver<?> resultReceiver, Row pa
             params,
             cursors,
             currentTransactionState,
-            new PlanStats(nodeCtx, txnCtx, tableStats)
+            new PlanStats(nodeCtx, txnCtx, tableStats),
+            planner::optimize
```

Comment:
I think as a follow up we could look into having a `planner.createContext(...)`, to keep the `optimize` internal and make the `PlannerContext` constructor package private.

But not sure of the implications, and if it will really improve anything but I think it might be worth a shot.